### DISCUSSION TITLE
Re-create related records on update from scratch

### DISCUSSION THREAD

**Discussion by medihack** [prisma/prisma#8519] [Q&A] [answered]
I want to re-create some related records on update, something like this:

    db.post.update({
      where: { id: 1 },
      data: {
        title: "The first post",
        comments: {
          set: [{ content: "foo"}, { content: "bar" }],
        },
      },
    })

This is unfortunately not possible as [`set`][1] only allows to reference existing comments (by using the `id`). So is there another way during an update to delete all those related comments and re-create them from scratch in one transaction (and maybe one nested write call)?

  [1]: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#set

**Comment by medihack** [comment, marked as answer]
According to  this [issue comment](https://github.com/prisma/prisma/issues/5456#issuecomment-772994699) it may be possible to do

```ts
db.post.update({
  where: { id: 1 },
  data: {
    title: "The first post",
    comments: {
      deleteMany: {},
      create: [{ content: "foo"}, { content: "bar" }],
    },
  },
})
```

I have not yet tried it myself, nor is it documented somewhere. Can one rely on this?

**Reply by ryands17** [reply]
Yes this is fine. The operations will run in the order that they are specified.

**Reply by medihack** [reply]
Thanks, good to know. But I still opened an issue #8522, because it would be nice to have it mentioned somewhere in the documentation at least or have an extra keyword (like `replace`) for it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
